src/dialect/onnx/onnx_ops.cpp

```diff
@@ -674,8 +679,23 @@ void ONNXReshapeOp::inferShapes() {
     if (valueAttribute.getValue().size() != outputRank)
       emitError("Constant value must have same rank as output");
 
-    for (int i=0; i<outputRank; ++i)
-      dims[i] = valueAttribute.getValue()[i].cast<IntegerAttr>().getInt();
+    int64_t numberOfDynamicInputs = 0;
+    int64_t totalKnownDimsSize = 1;
+    int64_t dynamicValueIndex = -1;
+    for (int i=0; i<outputRank; ++i) {
+      dims[i] = ArrayAttrIntVal(valueAttribute, i);
+      if (dims[i] < 0) {
```

Comment:
specs also say that if dim is zero, it keeps the same as the original; this seems to be missing